Automattic/wp-calypso
# Publicize: add sharing preview modal

This adds a preview modal to publicize:

`breakpoint > 660px `

![screen shot 2017-03-30 at 1 28 36 pm](https://cloud.githubusercontent.com/assets/744755/24524978/6c5aee22-154d-11e7-9fc7-ca98db854baa.png)

`breakpoint < 660px`

![screen shot 2017-03-30 at 1 29 05 pm](https://cloud.githubusercontent.com/assets/744755/24525195/4b141efe-154e-11e7-8a0b-16ff038e3cf2.png)

**Note** 
At the moment we only have previews for Facebook and Twitter. #12644 adds the remaining available services.

**Testing**
- Open sharing option on a post from http://calypso.localhost:3000/posts/my/:site
- Edit customized message
- Click "Preview"
- The preview modal should open
- The preview should show the customized message
- Click the close modal button ( top left X )
- The preview modal should close
